Moved all notebooks to docs/notebooks.

gnecula opened the pull request:
Now all notebooks are in the same place, thus all are subject
to auto-doc generation at readthedocs.io and to automated testing
with travis.

Some notebooks are too slow, exclude them at docs/conf.py:exclude_patterns.

Cleanup a bit the section headings in notebooks so that they show
up well in readtehdocs.io.

gnecula committed:
Moved all notebooks to docs/notebooks.

Now all notebooks are in the same place, thus all are subject
to auto-doc generation at readthedocs.io and to automated testing
with travis.

Some notebooks are too slow, exclude them at docs/conf.py:exclude_patterns.

Cleanup a bit the section headings in notebooks so that they show
up well in readtehdocs.io.

gnecula committed:
It seems that travis does not have 'sklearn'

jekbradbury commented:
It looks like the notebooks touched in this PR are deleted from `notebooks/` but not added to `docs/notebooks/`?

gnecula committed:
Added the lost notebooks to docs/notebooks

gnecula commented:
> It looks like the notebooks touched in this PR are deleted from `notebooks/` but not added to `docs/notebooks/`?

Good catch, added now

gnecula committed:
Merge remote-tracking branch 'upstream/master' into documentation

gnecula committed:
Increase the cell timeout for executing notebooks

Attempt to fix the Travis documentation testing

gnecula committed:
Exclude also the neural network notebook from auto-generation

It was timing out

gnecula committed:
Disable the score_matching notebook from auto-doc.

I did this because the notebook uses sklearn, which does not exist at Travis
Will file a separate issue with the notebook author to see if we can
remove the dependency.
